zipline/finance/performance/period.py

```diff
@@ -203,9 +200,12 @@ def calculate_performance(self):
 
     def record_order(self, order):
         if self.keep_orders:
-            dt_orders = self.orders_by_modified[order.dt]
-            if order.id in dt_orders:
-                del dt_orders[order.id]
+            try:
+                dt_orders = self.orders_by_modified[order.dt]
+                if order.id in dt_orders:
+                    del dt_orders[order.id]
```

Comment:
That's a good question. I was trying to keep existing logic, and merely change the lookup behavior.

But you're right, removing the del should be functionally equivalent. I'll tack on a patch addressing that.